File: 2DAM/ProgramacionDeServiciosYProcesos/Unidad-3/PrimeraConexionClienteServidor/src/main/java/es/drodriguez/com/Cliente.java
package es.drodriguez.com;

import java.io.DataInputStream;
import java.io.IOException;
import java.net.InetAddress;
import java.net.Socket;
import java.net.UnknownHostException;

public class Cliente {
    public static void main(String[] args) {
        System.out.println("CLIENTE");
        InetAddress direccion;
        Socket socket;
        int numeroCliente = 0;
        int PUERTO = 5000;

        System.out.println("Conectando con el servidor...");

        try{
            direccion = InetAddress.getLocalHost();
            socket = new Socket(direccion, PUERTO);
            DataInputStream entrada = new DataInputStream(socket.getInputStream());
            System.out.println("El servidor dice: " + entrada.readUTF());
            entrada.close();
            System.out.println("Se ha desconectado del servidor");


        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: 2DAM/ProgramacionDeServiciosYProcesos/Unidad-3/CalculadoraSocketKotlin/src/main/kotlin/Cliente.kt
import java.io.DataInputStream
import java.io.ObjectOutputStream
import java.net.InetAddress
import java.net.Socket

fun main() {
    println("CLIENTE CALCULATOR")
    val direccion: InetAddress
    val servidor: Socket
    val PUERTO: Int = 5656

    direccion = InetAddress.getLocalHost()
    servidor = Socket(direccion, PUERTO)

    //Obtenemos datos para empaquetar y enviar al servidor
    println("A continuación debe introducir los números y el tipo de operación que desea realizar")
    println("Introduce el número 1:")
    val numero1: Int? = readln().toIntOrNull()
    println("Introduce el número 2:")
    val numero2: Int? = readln().toIntOrNull()
    println("Introduce la operación que desea realizar:")
    println("SUMA/RESTA/MULTIPLICACION/DIVISION")
    val operacion: String? = readlnOrNull()

    try {
        require(numero1 != null && numero2 != null && operacion != null) { "No se ha introducido un número o la operación no es válida" }
        val paquete = Paquete(numero1, numero2, operacion)
        val salidaDatos = ObjectOutputStream(servidor.getOutputStream())
        salidaDatos.writeObject(paquete)


    } catch (e: Exception) {
        println(e.message)
    }
    //Recibimos el resultado del servidor
    val entradaDatos = DataInputStream(servidor.getInputStream())
    println(entradaDatos.readUTF())

}